test/OrchardCore.Tests/Apis/ContentManagement/DeploymentPlans/ContentStepLuceneQueryTests.cs

```diff
@@ -50,5 +56,109 @@ public async Task ShouldUpdateLuceneIndexesOnImport()
+                jItem[nameof(ContentItem.DisplayText)] = "new version";
+            });
+            const string secondcontentitemversionid = "secondcontentitemversionid";
+            var data = recipe["steps"][0]["Data"].AsArray();
+            var secondContentItem = JObject.FromObject(context.OriginalBlogPost);
+            secondContentItem[nameof(ContentItem.ContentItemVersionId)] = secondcontentitemversionid;
+            secondContentItem[nameof(ContentItem.DisplayText)] = "second content item version display text";
+            secondContentItem[nameof(AutoroutePart)][nameof(AutoroutePart.Path)] = "new-path";
+            data.Add(secondContentItem);
+
+            Assert.Equal(2, data.Count);
+            Assert.Single(data.Select(x => x.SelectNode(nameof(ContentItem.ContentItemId)).ToString()).Distinct());
+
+            // validate different contentItem versionId
+            var versionIds = data.Select(x => x.SelectNode(nameof(ContentItem.ContentItemVersionId)).ToString()).Distinct();
+            Assert.Equal(2, versionIds.Count());
+
+            await context.PostRecipeAsync(recipe);
+
+            // Search indexes are no longer updated in a deferred task at the end of a shell scope
+            // but in a background job after the http request, so they are not already up to date.
+
+            await context.UsingTenantScopeAsync(async scope =>
+            {
+
+
+                var luceneQuerySource = scope.ServiceProvider.GetRequiredService<LuceneQuerySource>();
+                var luceneQuery = new LuceneQuery
+                {
+                    Index = "Search",
+                    ReturnContentItems = false,
+                    Template = @"
+                            {
+                                ""query"": {
+                                    ""term"": { ""Content.ContentItem.ContentType"": ""BlogPost"" }
+                                },
+                                ""filter"": [
+                                    {
+                                        ""term"": { ""Content.ContentItem.Published"":""true"" }
+                                    },
+                                    {
+                                        ""term"": { ""Content.ContentItem.ContentItemId"":""{{parameters.contentItemId}}"" }
+                                    }
+                                ]
+                            }",
+                };
+                var queryParameters = new Dictionary<string, object>
+                {
+                    ["contentItemId"] = context.OriginalBlogPost.ContentItemId
+                };
+
+                var session = scope.ServiceProvider.GetRequiredService<YesSql.ISession>();
+                var checkIndexingTaskQuery = new SqlQuery
+                {
+                    Name = "CheckIndexingTask",
+                    Template = "Select Count(1) as Total from IndexingTask"
+                };
```

Comment:
@hishamco , Here , 
Looks like I found another Bug, and here it will never work because of syntax errors

https://github.com/OrchardCMS/OrchardCore/blob/c967b5d49d7bc85ae3d58f32aa45a94cfee4f872/src/OrchardCore.Modules/OrchardCore.Indexing/IndexingTaskManager.cs#L133-L141

Reply:
ADO.net directly use string collections as arguments in the `in` expressions?

Reply:
Apparently not, the logic here is only to prevent repeated insertions, and there is no code to clean up the index task after the task is completed 😢